subprojects/core/src/integTest/groovy/org/gradle/internal/operations/notify/BuildOperationNotificationIntegrationTest.groovy

```diff
@@ -135,26 +135,21 @@ class BuildOperationNotificationIntegrationTest extends AbstractIntegrationSpec
     }
 
     // This test simulates what the build scan plugin does.
-    def "can ignore buildSrc events by deferring registration"() {
+    def "listener receives recorded operations happen before registration"() {
```

Comment:
OK, I think it would be better to rely on a build that's simpler (doesn't use composite build or buildSrc) and look for operations that happen that way.  

And then separately, we should have a build for buildSrc/composite/continuous, but I think that makes sense as a separate step from this.